src/containers/error-boundary.jsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import platform from 'platform';
+import WebGlModalComponent from '../components/webgl-modal/webgl-modal.jsx';
+import log from '../lib/log.js';
+
+class ErrorBoundary extends React.Component {
+    constructor (props) {
+        super(props);
+        this.state = {
+            hasError: false
+        };
+    }
+
+    componentDidCatch (error, info) {
+        // Display fallback UI
+        this.setState({hasError: true});
+        log.error(`Unhandled Error: ${error}, info: ${info}`);
+    }
+
+    handleBack () {
+        window.history.back();
+    }
+
+    render () {
+        if (this.state.hasError) {
+            if (platform.name === 'IE') {
+                return <h1>Sorry Internet Explorer is not supported.</h1>;
+            }
+            if (window.WebGLRenderingContext) {
+                const canvas = document.createElement('canvas');
+                if (!canvas.getContext('webgl')) {
+                    return <WebGlModalComponent onBack={this.handleBack} />;
+                }
+            } else {
+                return <WebGlModalComponent onBack={this.handleBack} />;
+            }
+            return <h1>Something went wrong.</h1>;
```

Comment:
If this is just a placeholder, can you file an issue for discussion about what the generic "something terrible happened" action should be?

Reply:
👍  https://github.com/LLK/scratch-gui/issues/1205